SourceX/controls/controller_buttons.h

```diff
@@ -21,7 +21,12 @@ enum class ControllerButton {
 	BUTTON_LEFTSHOULDER,
 	BUTTON_RIGHTSHOULDER,
 	BUTTON_START,
-	BUTTON_BACK,
+	BUTTON_QUEST,

```

Comment:
This enum was designed to abstract the differences between SDL2, SDL1 Joystick, and devices that report their gamepad buttons as keyboard keys.

I think game-specific keys should live at a higher level (`GameAction`). My initial implementation of modifier key for RG300 wasn't done right in this regard.

Reply:
I would either need to call them BUTTON_A_MOD or implement an abstract IsModifierKey(), but that could get a bit herry. Do you have any suggestions for how to deal with this?

Reply:
ControllerButtons should remain low-level (no modifier buttons there).

Modifiers can be handled in the controls -> action mapping (e.g. `GetGameAction`). This will be less code, as modifier logic won't have to be duplicated in the 3 `devices` files.

> implement an abstract IsModifierKey()

This should be easy to do using `bool IsControllerButtonPressed(ControllerButton button)` from `controls/controller.h`.

Something like:

```c++
bool IsModifierKey() {
  return IsControllerButtonPressed(ControllerButton::BUTTON_BACK);
}
```

Reply:
Thanks for the hint, I'll update the PR sometime this week

Reply:
Ok reworked the PR, let me know how you feel about the changes :)

Reply:
Looks good to me!